app/controllers/users_controller.rb

```diff
@@ -0,0 +1,44 @@
+class UsersController < ApplicationController
+  before_action :authorize_request, except: :create
+  before_action :find_user, except: %i[create index]
+  def index
+    @users = User.all
+    render json: @users, status: :ok
+  end
+
+  def show
+    render json: @user, status: :ok
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      render json: @user, status: :created
+    else
+      render json: { errors: @user.errors.full_messages }, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    unless @user.update(user_params)
+      render json: { errors: @user.errors.full_messages },
+             status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @user.destroy
```

Comment:
You could send 200/202/204 status codes for this, kindly refer to  https://stackoverflow.com/questions/2342579/http-status-code-for-update-and-delete